internal-packages/workflow-designer-ui/src/settings/github-integration/github-integration-setting-form.tsx

```diff
@@ -103,53 +224,53 @@ function Installed({
 								</SelectContent>
 							</Select>
 						</fieldset>
-						<fieldset className="flex flex-col gap-[4px]">
-							<Label>Callsign</Label>
-							<input
-								type="text"
-								className="bg-black-750 h-[28px] border-[1px] border-white-950/10 flex items-center px-[12px] text-[12px] rounded-[8px] outline-none placeholder:text-white-400/70"
-								placeholder="Enter call sign"
-								name="callsign"
-								defaultValue={data?.callsign}
-							/>
-						</fieldset>
+						{requiresCallsign(
+							selectedTrigger as WorkspaceGitHubIntegrationTrigger,
+						) && (
+							<fieldset className="flex flex-col gap-[4px]">
+								<Label>Callsign</Label>
+								<input
+									type="text"
+									name="callsign"
+									className="bg-black-750 h-[28px] border-[1px] border-white-950/10 flex items-center px-[12px] text-[12px] rounded-[8px] outline-none placeholder:text-white-400/70"
+									value={callsign}
+									onChange={(e) => setCallsign(e.target.value)}
+								/>
+							</fieldset>
+						)}
 					</div>
 					<h3 className="font-accent text-white-400 text-[14px] font-bold">
 						Data mapping
 					</h3>
 					<div>
 						<PayloadMapForm
 							nodes={workspace.nodes}
-							currentPayloadMaps={data?.payloadMaps}
+							currentPayloadMaps={payloadMaps}
+							availablePayloadFields={availablePayloadFields}
 						/>
 					</div>
 					<h3 className="font-accent text-white-400 text-[14px] font-bold">
 						Then
 					</h3>
 					<fieldset className="flex flex-col gap-[4px]">
-						<Select name="nextAction" defaultValue={data?.nextAction}>
+						<Select
+							name="nextAction"
+							value={selectedNextAction}
+							onValueChange={(value) =>
+								setSelectedNextAction(
+									value as WorkspaceGitHubNextIntegrationAction,
```

Comment:
import `WorkspaceGitHubIntegrationNextAction` then
```suggestion
									WorkspaceGitHubIntegrationNextAction.parse(value),
```